src/pages/guides/using/idcs/idcs-events-data-stream-setup.md

```diff
@@ -0,0 +1,223 @@
+
+## Setup events browsing/listning in IDCS
+
+InDesign cloud services is currently a Beta Product and thus cannot be enabled without a PreRelease agreement. After a PreRelease agreement is in place, you should be able to see InDesign cloud services events tile in developer console.
+
+## Setup Adobe I/O
+
+See [Getting Started with Adobe I/O Events](/src/pages/index.md)
+
+For basic instructions for this use case, starting from [developer.adobe.com/console](/console/):
+
+*When prompted, click the designated button to proceed*
+
+- Select `Create new project`
+
+  ![Create new project](../../img/events_idcs_01.png "Quick Start")
+
+- Select `Add event`
+
+  ![Add event](../../img/events_idcs_02.png "Get started with your new project by adding an event subscription")
+
+- Filter by `Creative Cloud`
+  ![Filter Selection](../../img/events_idcs_03.png "Filter by Creative Cloud")
+
+- Select `Select InDesign Cloud Services`
+
+  ![Provider selection](../../img/events_idcs_04.png "Select event provider")
+
+- Subscribe to the user driven change events of your choosing
+
+  ![Event selection](../../img/events_idcs_05.png "Select event subscriptions")
+
+- Set up OAuth Server-to-Server Credentials.
+  - The OAuth Server-to-Server credential relies on the OAuth 2.0 client_credentials grant type to generate access tokens.
+
+  ![Set up credentials](../../img/events_idcs_06.png "Set up credentials")
+
+- Set up Event Registration
+  - Provide a name and description for this event subscription
+
+  ![Set Up registration](../../img/events_idcs_07.png "Set Up registration")
+  
+- Configure Event Registration
+
+  ![Configure registration](../../img/events_idcs_08.png "Configure registration")
+
+  - Optionally choose whether to enable Webhook or Runtime action
+    - Enable Webhook
+      - We recommend batch over single webhooks
+      - For `Webhook URL` a public https endpoint must be provided
+      - The endpoint must be able to handle get and post requests
+      - The get request must respond with the challenge query if it exists
+      - The post request must respond that it received the message or the webhook will re-attempt to send several times before giving up and automatically disabling the webhook sends
+    - Enable Runtime action
+      - [See Setting up your Runtime Environment](/runtime/docs/guides/getting-started/)
+      - Select a pre-made runtime action/runtime namespace
+- After Saving
+
+  ![Verify setup](../../img/events_idcs_09.png "Verify setup")
+  
+  - Verify that the Status is `Active`
+  - If Webhook was selected, verify that it successfully passed the challenge without errors
+
+### Event Data Structure
+
+*We are still finalizing the data structure as we prepare for Beta release. While we don't expect any major changes, there may be some minor modifications.*
+
+Events are structured in JSON format using the [CloudEvents](https://cloudevents.io/) spec
+
+*Example Event (batch)*
+
+````json
+
+{
+  "events": [
```

Comment:
This will not be wrapped in this object. A batch delivery would be a simple json array delivery. For example:
```json
[
  {event1},
  {event2}
]

Reply:
Updated.